### PR TITLE
Adjust handshake mode on retransmission.

### PR DESCRIPTION
**Pull request by boaks**
Signed-off-by: Achim Kraus <achim.kraus@bosch.io>

### CODE REVIEW

**Comment by boaks** [comment]
@sbernard31 

check, if you want the fix in 2.2.0. For me it will be also OK, to include that after the 2.2.0.

**Comment by sbernard31** [comment]
Just to know, what could be the reason to not integrate it ? you consider you don't test it enough ?

**Comment by boaks** [comment]
> you consider you don't test it enough ?

Yes, because of the short time. And it now changes the `Request`'s `EndpointContext` itself. I assume, this will be not too often effective, because the HandshakeModes are rarely used.

**Comment by sbernard31** [comment]
Just to be sure this fix is only needed when default handshake mode is AUTO ?

I mean in Leshan we set a default handshake mode to NONE and set manually on the request  `HANDSHAKE_MODE_AUTO` when needed ?

**Comment by boaks** [comment]
It keeps NONE and remove the others. If leshan uses only NONE, we don't need it for 2.2.0.

**Comment by sbernard31** [comment]
:+1:

File: element-connector/src/main/java/org/eclipse/californium/elements/EndpointContextUtil.java [hunk 1]
```diff
@@ -61,4 +61,33 @@ public static boolean match(String name, Set<String> keys, EndpointContext conte
...
+	 * 
+	 * @param messageContext messages's endpoint context.
+	 * @param connectionContext the connection's endpoint context. Either
+	 *            reported with
+	 *            {@link RawData#onContextEstablished(EndpointContext)} or
+	 *            contained in a received message.
+	 * @return endpoint context to be used for follow-up messages.
+	 * @since 2.3
+	 */
+	public static EndpointContext getFollowUpEndpointContext(EndpointContext messageContext,
```
Comment (sbernard31) [inline element-connector/src/main/java/org/eclipse/californium/elements/EndpointContextUtil.java:80]: clearly a detail but I would change the name to make clear this is for block. (even if this is already mention in the javadoc)

Comment (boaks) [reply element-connector/src/main/java/org/eclipse/californium/elements/EndpointContextUtil.java:80]: It's also for "retransmissions of CONs". 
So I'm not sure, which name you have in mind?

Comment (sbernard31) [reply element-connector/src/main/java/org/eclipse/californium/elements/EndpointContextUtil.java:80]: correct forget it !

**Pull request merged**